src/perovskite_solar_cell_database/llm_extraction_schema.py

```diff
@@ -486,5 +479,25 @@ class LLMExtractedPerovskiteSolarCell(PublicationReference, SectionRevision, Sch
         section_def=Layer, repeats=True, a_eln=ELNAnnotation(label='Layers')
     )
 
+    layer_order = Quantity(
+        type=str,
+        description='Order of the layers in the device stack. Use the layer names as they appear in the "Layers" section, separated by commas.',
+        a_eln=ELNAnnotation(label='Layer Order', component='StringEditQuantity'),
+    )
+
+    # normalizer that reorderes the layers according to the layer_order
+    def normalize(self, archive, logger):
+        if not self.layer_order:
+            return
+
+        layer_dict = {layer.name: layer for layer in self.layers}
+        ordered_names = [name.strip() for name in self.layer_order.split(',')]
+
+        if set(ordered_names) != set(layer_dict.keys()):
+            raise ValueError('Layer order does not match available layers')
```

Comment:
I think it would be better to log a warning and return. Something like:
```suggestion
            logger.warn('The names in layer_order does not match available layers')
            return
```